tox.ini

```diff
@@ -30,5 +30,14 @@ commands = {posargs}
 # W504 ignore wrap line after binary operator
 ignore = E501,W504
 show-source = True
-exclude = ./.*,build,dist,*egg
+exclude = ./.*,build,dist,*egg,venv
 import-order-style = pep8
+
+[testenv:coverage]
+description = "Checking the test coverage of the code."
+deps = coverage
+commands =
+  coverage erase
+  coverage run --branch --include='git_machete/*' --omit='git_machete/tests/*' setup.py test
```

Comment:
Hmmm for clarity... against which Python version will this test be done? 🤔 Since this `coverage` env is distinct from `py36-py39`.

Actually, I now realize the same point applies to `pep8` tox env... if you could check 😅

Reply:
I have checked `python --version` and `python3 --version`, since there I haven't found any explicit way to verify python version through flake8. I assume that default python is one that is being used since we don't specify anything else there. We could change flake8 to `python{PYTHON_VERSION} -m flake8` and `python{PYTHON_VERSION} -m coverage...`, so we explicitly run it using particular python.

py 3.6 - pep is run using python 3.6.14 as it is default python
py 3.7 - pep is run using 3.7.11 as it is default python
py 3.8 - pep is run using 3.8.11 as it is default python
py 3.9 - pep and coverage are run using 3.9.6 as it is default python

Reply:
Okay, sorry, I wasn't clear about what I mean here (or rather, I didn't think the question over enough):
* in CI setup, it's rather clear which Python version is used &mdash; the one that's present in the given step's Docker image
* but! in local setup, while for py36-py39 it's clear which Python version is used... which version of Python is used for `pep8` and `coverage`? Is it just whatever version is linked to `python` (`python3`?) executable in the system?

Reply:
Both coverage and pep8 are run through tox, so I would expect that they're run using whatever is running tox. So if you just run `tox -e coverage` you will use python on which you installed tox. If there are more than one than you can check `tox --showconfig` and there are parameters in [tox] section - host_python and in [testenv:coverage] - basepython. You can also specify base_python for particular testenv.

Reply:
Ok! thanks, that explains it to me